libcontainer/container_linux.go

```diff
@@ -823,23 +805,24 @@ func (c *linuxContainer) currentState() (*State, error) {
 func (c *linuxContainer) orderNamespacePaths(namespaces map[configs.NamespaceType]string, doInit bool) ([]string, error) {
 	paths := []string{}
 	nsTypes := []configs.NamespaceType{
+		configs.NEWUSER,
 		configs.NEWIPC,
 		configs.NEWUTS,
 		configs.NEWNET,
 		configs.NEWPID,
 		configs.NEWNS,
 	}
-	// For now, only join user namespace if this is an exec in process and the
-	// container supports user namespace
-	if !doInit && c.config.Namespaces.Contains(configs.NEWUSER) {
-		nsTypes = append(nsTypes, configs.NEWUSER)
-	}
```

Comment:
You mix joining and creating namespaces. Let's remember the fact that when you attach or create userns, you lose privileges.

If you have a list of namespaces which should be created, you need to create a userns first and than other namespaces. In this case you will have full priveleges in new namespaces.

But if you are going to attach to namespaces, you need to attach to all namespaces except userns and only then attach to the userns. Because you may have insufficient privilages to attach other namespaces from the userns.